grblas/matrix.py

```diff
@@ -22,20 +22,43 @@ def __repr__(self):
         return f'<Matrix {self.nvals}/({self.nrows}x{self.ncols}):{self.dtype.name}>'
 
     def __eq__(self, other):
-        # Borrowed this recipe from LAGraph
+        return self.isequal(other)
+
+    def isequal(self, other, rel_tol=1e-7, abs_tol=0.0, strict_dtype=False):
```

Comment:
I'm not a fan of defaulting `isequal` to do `isclose`.  Equals means equals to me and should correspond to a single GraphBLASS call.  Why not also have `isclose`?  In numpy, `isclose` is element-wise and `allclose` returns boolean True or False, so I suppose there could be some confusion in behavior of a like-named function.

I've probably asked this before, but what do you think about adding element-wise operations such as `__lt__`?  It begs the question whether `ewise_add` or `ewise_mult` should be used, but `ewise_mult` is the only one that makes sense.  Defining `__eq__` as above that returns boolean would make adding these in the future more difficult.  Adding element-wise operations as conveniences also impacts our name of `isclose` or `allclose` if we were to go that route.

I'm curious whether and when the non-blocking version of SS:GraphBLAS will be able to fuse the operations used for the tolerance calculation and then short-circuit when a False is encountered during the `LAND` reduction.  Actually, never mind, we need to calculate the number of values from the comparison, so we can't delay.  Relatedly, I'm curious how a numba function to perform the tolerance calculation would perform.

I'm also split on using `rel_tol=` and `abs_tol=` vs `rtol=` and `atol=`.  The former is used by `math.isclose` and the latter by `numpy.isclose`.  Where did you come up with the default tolerance values?  I slightly favor `rtol` and `atol` (probably because I'm more familiar with numpy).

I'm not crazy about `strict_dtype=` either, but I can live with it.  `A.dtype == B.dtype and A.isequal(B)` isn't that bad, but neither is `A.isequal(B, strict_dtype=True)`.  Also, `strict_dtype` should be keyword-only IMHO.

Reply:
Per the discussion, let's go with:
`A.isequal(B, *, check_dtype=False)`
`A.isclose(B, rtol=1e-7, atol=0.0, *, check_dtype=False)`
Remove `==`

Reply:
I've been going back and forth on `numpy`-like vs `math`-like.  Right now I prefer:

`A.isclose(B, *, rel_tol=1e-9, abs_tol=0.0, check_dtype=False)`

which matches the signature of `math.isclose`.